### ISSUE TITLE
Incorrect certificate chain used in SslStream_UntrustedCaWithCustomTrust_OK test.

### ISSUE THREAD

**Issue by rzikm** [dotnet/runtime#73295]
Discovered while working on https://github.com/dotnet/runtime/issues/72873.

During the original implementation, there was a typo at 

https://github.com/dotnet/runtime/blob/cfa4f6f26e303eb0f0f67e3798fcf005e0124092/src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs#L791

From discussion with @wfurt, the second argument should be `serverChain`. However, the change breaks tests on Mac so more investigation is needed:
```
    System.Net.Security.Tests.SslStreamNetworkStreamTest.SslStream_UntrustedCaWithCustomTrust_OK(usePartialChain: False) [FAIL]
      System.Security.Authentication.AuthenticationException : The remote certificate is invalid because of errors in the certificate chain: PartialChain
      Stack Trace:
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs(477,0): at System.Net.Security.SslStream.SendAuthResetSignal(ProtocolToken message, ExceptionDispatchInfo exception)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs(529,0): at System.Net.Security.SslStream.CompleteHandshake(SslAuthenticationOptions sslAuthenticationOptions)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs(332,0): at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](Boolean receiveFirst, Byte[] reAuthenticationData, CancellationToken cancellationToken)
        /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs(120,0): at System.Threading.Tasks.TaskTimeoutExtensions.GetRealException(Task task)
        --- End of stack trace from previous location ---
        /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs(90,0): at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks)
        /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs(55,0): at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout)
        /_/src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs(805,0): at System.Net.Security.Tests.SslStreamNetworkStreamTest.SslStream_UntrustedCaWithCustomTrust_OK(Boolean usePartialChain)
        --- End of stack trace from previous location ---
```

**Comment by wfurt** [comment]
The test got broken by https://github.com/dotnet/runtime/pull/47729 in 6.0. Originally introduced by  #46664.

**Comment by karelz** [comment]
Moving to 8.0 due to lack of time as per above.

**Comment by wfurt** [comment]
This was primarily fixed by #73577. I was waiting for #73745 to get more test coverage.
Remaining (test only ) issue is tracked by https://github.com/dotnet/runtime/issues/73862
